CatalogReaderExample/NuGet.Protocol.Catalog/CatalogProcessorSettings.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Protocol.Catalog
+{
+    /// <summary>
+    /// Settings for how <see cref="CatalogProcessor" /> should behave. Defaults to processing all catalog items on
+    /// <see cref="https://api.nuget.org/v3/index.json"/>.
+    /// </summary>
+    public class CatalogProcessorSettings
+    {
+        public CatalogProcessorSettings()
+        {
+            ServiceIndexUrl = "https://api.nuget.org/v3/index.json";
+            DefaultMinCommitTimestamp = null;
+            MinCommitTimestamp = DateTimeOffset.MinValue;
+            MaxCommitTimestamp = DateTimeOffset.MaxValue;
+            ExcludeRedundantLeaves = true;
```

Comment:
a settings class shouldn't have defaults. It's confusing. Better to set explicitly in the code entry point.

Reply:
why is it confusing?

what would you consider the "code entry point" in this situation?

Reply:
I considered this, but I wanted it to be as simple as possible to start hacking against the catalog. I think the most likely case is that people will be running this against NuGet.org and will care about all events. That's why I set the defaults.

Reply:
Code entry point in this case is main(). Don't feel strongly about this one.

Reply:
Kept with defaults but added tests so the defaults don't change on people.